Calendar/BookingHelpers/GraphInstaller.ps1

```diff
@@ -0,0 +1,84 @@
+﻿# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# ===================================================================================================
+# Graph Modules installer Functions
+# ===================================================================================================
+function CheckGraphBookingsModuleInstalled {
+    [CmdletBinding(SupportsShouldProcess=$true)]
+    param ()
+
+    if (Get-Command -Module Microsoft.Graph.Bookings) {
+        Write-Host "Microsoft.Graph.Bookings module is already installed."
+    } else {
+        # This is slow, to the tune of ~10 seconds, but much more complete.
+        # Check if ImportExcel module is installed
+        $moduleInstalled = Get-Module -ListAvailable | Where-Object { $_.Name -eq 'Microsoft.Graph.Bookings' }
+
+        if ($moduleInstalled) {
+            Write-Host "Microsoft.Graph.Bookings module is already installed."
+        } else {
+            # Check if running with administrator rights
+            $isAdministrator = ([Security.Principal.WindowsPrincipal] [Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole]::Administrator)
+
+            if (-not $isAdministrator) {
+                Write-Host "Please run the script as an administrator to install the Microsoft.Graph.Bookings module."
+                exit
+            }
+
+            # Ask user if they want to install the module
+            if ($PSCmdlet.ShouldProcess('Microsoft.Graph.Bookings module', 'Import')) {
+                Write-Verbose "Installing Microsoft.Graph.Bookings module..."
+
+                # Install ImportExcel module
+                Install-Module -Name Microsoft.Graph.Bookings -Force -AllowClobber
+
+                Write-Host "Done. Microsoft.Graph.Bookings module is now installed. Please re-run the script."
+                exit
```

Comment:
>exit [](http://example.com/codeflow?start=16&length=4)

Should it CheckGraphAuthModuleInstalled before exiting?
i.e. install both at once.

Reply:
cannot check for the 3 modules, including ImportExcel, as Excel is a shared function and has its own exit